tests/test-zmq-endpoints.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# This script sets up 2 nmsgtools: A ZMQ listener and a ZMQ writer.  The
+# writer attempts to write $entry_count entries to a ZMQ socket which the listener
+# is listening on.  If, at the end of the test, we do not have the correct
+# number of entries in the listener's output file, then we fail.
+
+script_dir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+nmsgtool_test=$script_dir/../src/nmsgtool
+infile=$script_dir/private-tests/202_sample.nmsg
+outfile=/tmp/testzmq.json
+sock=/tmp/testzmq.sock
+entry_count=15
+retval=0
+
+check() {
+	if [ $? = "0" ]; then
+		echo "PASS: $*"
+	else
+		echo "FAIL: $*"
+		retval=1
+	fi
+}
+
+echo "Testing ZMQ socket connection: "
+
+# Create a listener and a writer on the same socket.
+$nmsgtool_test -ddddd --unbuffered -J $outfile -L ipc://$sock,pushpull,connect &
+listener_pid=$!
+$nmsgtool_test -ddddd --unbuffered -r $infile -S ipc:///tmp/testzmq.sock,pushpull,accept &
+writer_pid=$!
+echo Listening PID: $listener_pid, writing PID: $writer_pid
+
+# Wait for the messages to send/receive.
+sleep 1
+
+# Kill the nmsgtools
+kill $writer_pid >/dev/null 2>&1
+killwriter_pid=$!
```

Comment:
More places where there is unnecessary waiting on the kill command

Reply:
Fixed